src/win/fs-event.c

```diff
@@ -97,12 +98,16 @@ static int uv_split_path(const WCHAR* filename, WCHAR** dir,
 
   if (i == 0) {
     if (dir) {
-      *dir = (WCHAR*)uv__malloc((MAX_PATH + 1) * sizeof(WCHAR));
+      dir_len = GetCurrentDirectoryW(0, NULL);
+      if (dir_len == 0) {
+        return -1;
+      }
+      *dir = (WCHAR*)uv__malloc((dir_len + 1) * sizeof(WCHAR));
```

Comment:
Adding 1 here is apparently not needed as the size returned by GetCurrentDirectoryW() already accounts for the terminating null, from https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getcurrentdirectory: 
_If the buffer that is pointed to by lpBuffer is not large enough, the return value specifies the required size of the buffer, in characters, including the null-terminating character_

Reply:
Ok!